client/dashboard/domains/domain-forwardings/delete-modal.tsx

```diff
@@ -0,0 +1,82 @@
+import { domainForwardingDeleteMutation } from '@automattic/api-queries';
+import { useMutation } from '@tanstack/react-query';
+import {
+	__experimentalText as Text,
+	__experimentalVStack as VStack,
+	Button,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useAnalytics } from '../../app/analytics';
+import { ButtonStack } from '../../components/button-stack';
+import type { DomainForwarding } from '@automattic/api-core';
+
+interface DomainForwardingDeleteModalProps {
+	domainName: string;
+	domainForwarding: DomainForwarding;
+	onClose?: () => void;
+}
+
+const DomainForwardingDeleteModal = ( {
+	domainName,
+	domainForwarding,
+	onClose,
+}: DomainForwardingDeleteModalProps ) => {
+	const deleteMutation = useMutation( domainForwardingDeleteMutation( domainName ) );
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+	const { recordTracksEvent } = useAnalytics();
+
+	const onConfirm = () => {
+		deleteMutation.mutate( domainForwarding.domain_redirect_id, {
+			onSuccess: () => {
+				createSuccessNotice( __( 'Domain forwarding rule was deleted successfully.' ), {
+					type: 'snackbar',
+				} );
+
+				recordTracksEvent( 'calypso_dashboard_domain_forwardings_delete_record', {
```

Comment:
I think `forwardings` is spelled incorrectly. I can see that it's also misspelled in the main component, but we should probably fix that as well in another PR. WDYT?